File: 2444.py
class Solution:
    def countSubarrays(self, nums: List[int], minK: int, maxK: int) -> int:
        res = 0
        lastMinKIdx = -1
        lastMaxKIdx = -1
        lastObIdx = -1
        for e in range(len(nums)):
            if not (minK <= nums[e] <= maxK):
                lastObIdx = e
            else:
                if nums[e] == minK:
                    lastMinKIdx = e
                if nums[e] == maxK:
                    lastMaxKIdx = e
                res += max(0, min(lastMinKIdx, lastMaxKIdx) - lastObIdx)
        return res

File: sol/2444.cpp
struct Solution {
    long long countSubarrays(vector<int>& nums, int minK, int maxK) {
        long long res = 0;
        int lastMinK = -1, lastMaxK = -1, lastOb = -1;
        for (int e = 0; e < nums.size(); ++e){
            if (minK > nums[e] || nums[e] > maxK){
                lastOb = e;
            } else {
                if (nums[e] == minK){
                    lastMinK = e;
                }
                if (nums[e] == maxK){
                    lastMaxK = e;
                }
                res += max(0, min(lastMinK, lastMaxK) - lastOb);
            }
        }
        return res;
    }
};
